File: engine/src/main/java/org/jmedina/jtetris/engine/model/BoardOperation.java
package org.jmedina.jtetris.engine.model;

import java.util.List;

import org.jmedina.jtetris.engine.enumeration.BoardOperationEnumeration;
import org.jmedina.jtetris.engine.figure.Box;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Jorge Medina
 *
 */
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class BoardOperation {

	private BoardOperationEnumeration operation;
	private List<Box> boxes;
	private long timeStamp;
}

File: frontend/src/app/model/board/boardOperation.ts
import { Box } from '../figure/box';

export type BoardOperationType =
  | 'BOARD_WITH_ADDED_FIGURE'
  | 'BOARD_WITH_OCURRED_ONE_LINE'
  | 'BOARD_WITH_OCURRED_TWO_LINES'
  | 'BOARD_WITH_OCURRED_THREE_LINES'
  | 'BOARD_WITH_OCURRED_FOUR_LINES'
  | undefined;

export interface BoardOperation {
  boxes?: Box[];
  timeStamp: number;
  operation: BoardOperationType;
}
